book/src/simple-payment-and-state-verification.md

```diff
@@ -0,0 +1,180 @@
+state receipt would point to the same value for A or B.
+
+The bank-merkle is computed from the merkle tree of the new state changes, along
+with the previous bank-merkle, and the block-merkle.
+ 
+```                
+                        bank-merkle 
+                   /                \
+                  /                  \
+          bank-diff-merkle        block-merkle
+                /       \
+               /         \
+            hash          previous bank-diff-merkle
+     /             \                   
+    /               \                   
+hash(Account1)  hash(Account2) 
+
+```
+
+A state receipt contains only the updated state changes for the block.  A direct
+Merkle path to the current bank-merkle guarantees the state value at that bank
+hash, but it cannot be used to generate a “current” receipt to the latest state
+if the state modification occurred in some previous block.  There is no
+guarantee that the path provided by the validator is the latest one available
+out of all the previous bank-merkles.
+
+Clients that want to query the chain for a receipt of the "latest" state would
+need to create a transaction that would update the Merkle path for that account,
+such as a credit of 0 lamports.
+
+###  Validator Votes
+
+Leaders should coalesce the validator votes by stake weight into a single entry.
+This will reduce the number of entries necessary to create a receipt.
+
+###  Chain of Entries
+
+A receipt is a PoH link from the payment or state Merkle path to a consecutive
+list of validation votes.
+
+
+It contains the following:
+* State -> Bank-Merkle
+or 
+* Transaction -> Entry-Merkle -> Block-Merkle -> Bank-Merkle
+
+And a vector of PoH entries:
+
+* Validator vote entries
+ 
+```
+* Ticks
+* Light entries 
+
+```
+/// This Entry definition skips over the transactions and only contains the
+/// hash of the transactions used to modify PoH.
+LightEntry { 
+    /// The number of hashes since the previous Entry ID.
+    pub num_hashes: u64,
+    /// The SHA-256 hash `num_hashes` after the previous Entry ID.
+    hash: Hash,
+    /// The Merkle root of the transactions encoded into the Entry.
+    entry_hash: Hash,
+}
+```
+
+The compressed entries are reconstructed from Entries and simply show the entry
+Merkle root that was used to modify PoH, instead of the full transaction set.
+
+Clients do not need the starting vote state. The [fork
+selection](book/src/fork-selection.md) algorithm is defined such that only votes
+that appear after the transaction provide finality for the transaction, and
+finality is independent of the starting state.
+
+###  Verification
+
+A light client that is aware of the supermajority set validators can verify a
+receipt by following the Merkle path to the PoH chain.  The Bank-Merkle is the
+Merkle root and will appear in votes included in an Entry.  The light client can
+simulate [fork selection](book/src/fork-selection.md) for the consecutive votes
```

Comment:
Is there any data available regarding resource usage for fork selection?

Reply:
It’s really light weight.  Just stake weighted timeouts. Should be on the order of number of nodes.

Reply:
But, we would likely need something more sophisticated, like shnor or bls signatures to actually be able to do SPV bidirectionally with eth.

Reply:
For signature aggregation, you mean?